internal/compiler/langtype.rs

```diff
@@ -764,20 +771,46 @@ impl EnumerationValue {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct LengthConversionPowers {
+    pub rem_to_px_power: Option<i8>,
+    pub px_to_phx_power: Option<i8>,
+}
+
 /// If the `Type::UnitProduct(a)` can be converted to `Type::UnitProduct(a)` by multiplying
-/// by the scale factor, return that scale factor, otherwise, return None
-pub fn unit_product_length_conversion(a: &[(Unit, i8)], b: &[(Unit, i8)]) -> Option<i8> {
+/// by the window scale factor and the default font size, return the scale factors,
+/// otherwise, return None
+pub fn unit_product_length_conversion(
+    a: &[(Unit, i8)],
+    b: &[(Unit, i8)],
+) -> Option<LengthConversionPowers> {
     let mut it1 = a.iter();
     let mut it2 = b.iter();
     let (mut v1, mut v2) = (it1.next(), it2.next());
     let mut ppx = 0;
     let mut lpx = 0;
+    let mut font_px = 0;
+    let mut font_rem = 0;
     loop {
         match (v1, v2) {
-            (None, None) => return (ppx == -lpx && ppx != 0).then(|| ppx),
+            (None, None) => {
+                return {
+                    Some(LengthConversionPowers {
```

Comment:
You should return Some here only if there is a conversion to do  (the calling code checks that)